src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,112 @@
+            getManufacturerStatement.setLong(1, id);
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(handleManufacturer(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer from DB by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String selectRequest = "SELECT * FROM manufacturers WHERE is_deleted = FALSE;";
+        List<Manufacturer> allManufacturers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllManufacturersStatement
+                        = connection.prepareStatement(selectRequest)) {
+            ResultSet resultSet = getAllManufacturersStatement.executeQuery(selectRequest);
+            if (resultSet.next()) {
+                allManufacturers.add(handleManufacturer(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufacturers from DB", e);
+        }
+        return allManufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateRequest = "UPDATE manufacturers SET name = ?, country = ? "
+                + "WHERE is_deleted = FALSE AND id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateManufacturerStatement
+                        = connection.prepareStatement(updateRequest)) {
+            updateManufacturerStatement.setString(1, manufacturer.getName());
+            updateManufacturerStatement.setString(2, manufacturer.getCountry());
+            updateManufacturerStatement.setLong(3, manufacturer.getId());
+            updateManufacturerStatement.executeUpdate();
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer name "
+                    + manufacturer.getName() + ", manufacturer country "
+                    + manufacturer.getCountry(), e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE manufacturers SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteManufacturerStatement
+                        = connection.prepareStatement(deleteRequest)) {
+            deleteManufacturerStatement.setLong(1, id);
+            return deleteManufacturerStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturer from DB by id " + id, e);
+        }
+    }
+
+    private static Manufacturer handleManufacturer(ResultSet resultSet) throws SQLException {
```

Comment:
The method should not be a static
```suggestion
    private Manufacturer handleManufacturer(ResultSet resultSet) throws SQLException {
```